TM1py/Services/ElementService.py

```diff
@@ -73,12 +75,35 @@ def delete(self, dimension_name: str, hierarchy_name: str, element_name: str, **
             element_name)
         return self._rest.DELETE(url, **kwargs)
 
-    def delete_elements(self, dimension_name: str, hierarchy_name: str, elements: str= None, **kwargs):
+    @require_version("11.4")
+    def delete_elements(self, dimension_name: str, hierarchy_name: str, element_names: List[str] = None,
+                        use_ti: bool = False, **kwargs):
+        if use_ti:
+            return self.delete_elements_use_ti(dimension_name, hierarchy_name, element_names, **kwargs)
+
         h_service = self._get_hierarchy_service()
-        h = h_service.get(dimension_name, hierarchy_name)
-        for ele in elements:
+        h = h_service.get(dimension_name, hierarchy_name, **kwargs)
+        for ele in element_names:
             h.remove_element(ele)
-        h_service.update(h)
+        h_service.update(h, **kwargs)
+
+    def delete_elements_use_ti(self, dimension_name: str, hierarchy_name: str, element_names: List[str] = None,
+                               **kwargs):
+        subset_service = self._get_subset_service()
+        unbound_process_name = subset_name = self.suggest_unique_object_name()
+        subset = Subset(subset_name, dimension_name, hierarchy_name, elements=element_names)
+        subset_service.update_or_create(subset, private=False, **kwargs)
+        try:
+            process_service = self._get_process_service()
+            process = Process(
+                name=unbound_process_name,
+                prolog_procedure=f"HierarchyDeleteElements('{dimension_name}', '{hierarchy_name}', '{subset_name}');")
+            success, status, error_log_file = process_service.execute_process_with_return(process, **kwargs)
+            if not success:
+                raise TM1pyException(f"Failed to clear elements through unbound process. Error: '{error_log_file}'")
```

Comment:
Should the error message say failed to clear or failed to delete/remove?

Reply:
Good catch. Thanks